File: ERP/HomeFloory/Entities/Proizvod.cs
using System;
using System.Collections.Generic;

namespace HomeFloory.Models;

public partial class Proizvod
{
    public decimal IdProizvod { get; set; }

    public string? Naziv { get; set; }

    public string? Opis { get; set; }

    public decimal? KolicinaNaStanju { get; set; }

    public decimal? CenaPoM2 { get; set; }

    public decimal? PaketPoM2 { get; set; }

    public string? Dimenzija { get; set; }

    public string? Nijansa { get; set; }

    public string? UrlSlike { get; set; }

    public decimal IdKategorija { get; set; }

    public decimal? IdProizvodjac { get; set; }

    public virtual ICollection<DodatiProizvodi> DodatiProizvodi { get; } = new List<DodatiProizvodi>();

    public virtual Kategorija IdKategorijaNavigation { get; set; } = null!;

    public virtual Proizvodjac? IdProizvodjacNavigation { get; set; }
}

File: Frontend/src/app/shared/models/proizvod.ts
export interface Proizvod {
    idProizvod: number
    naziv: string
    opis: string
    kolicinaNaStanju: number
    cenaPoM2: number
    paketPoM2: number
    dimenzija: string
    nijansa: string
    urlSlike: string
    idKategorija: number
    idProizvodjac?: number
    dodatiProizvodi?: any[]
    idKategorijaNavigation?: IdKategorijaNavigation
    idProizvodjacNavigation?: IdProizvodjacNavigation
}

export interface ProizvodAdd {
    naziv: string
    opis: string
    kolicinaNaStanju: number
    cenaPoM2: number
    paketPoM2: number
    dimenzija: string
    nijansa: string
    urlSlike: string
    idKategorija: number
    idProizvodjac?: number
}


export interface IdKategorijaNavigation {
    idKategorija: number
    nazivKategorije: string
    proizvodi: Proizvod | undefined[]
}

export interface IdProizvodjacNavigation {
    idProizvodjac: number
    nazivProizvodjaca: string
    proizvodi: Proizvod | undefined[]
}
